CMakeLists.txt

```diff
@@ -169,8 +169,8 @@ if (APPLE)
 else (APPLE)
    if (MINGW)
       # -mno-ms-bitfields see #22048
-      set(CMAKE_CXX_FLAGS_DEBUG   "-std=c++11 -mno-ms-bitfields -g")
-      set(CMAKE_CXX_FLAGS_RELEASE "-std=c++11 -mno-ms-bitfields -O2 -DNDEBUG -DQT_NO_DEBUG")
+      set(CMAKE_CXX_FLAGS_DEBUG   "-std=gnu++11 -mno-ms-bitfields -g")
+      set(CMAKE_CXX_FLAGS_RELEASE "-std=gnu++11 -mno-ms-bitfields -O2 -DNDEBUG -DQT_NO_DEBUG")
```

Comment:
This change seems to result in several new warnings. Not necessarily a bad thing ;-) but something should be done about those

Reply:
Without this change, the precompiled header doesn't work with Qt 5.10. Cmake automatically adds this flag when using Qt but the flag is not added to the compilation of the precompiled header. If the flags are different the PCH is  ignored.

Reply:
Hmm, I don't see these warning when using that change with Qt 5.8.x here.

Reply:
Yes, just those. And only on AppVeyour, locally it is just 2 warnings one unused function and one unused parameter

Reply:
Ok nothing severe.